core/src/main/java/hudson/model/AbstractCIBase.java

```diff
@@ -131,8 +137,8 @@ private void updateComputer(Node n, Map<String,Computer> byNameMap, Set<Computer
                     }
                 }
             }
+            used.add(c);
```

Comment:
I suspect it deserves a Jenkins issue for proper documentation/annotation, but it is not a part of this PR IMHO.